qiskit_optimization/applications/graph_optimization_application.py

```diff
@@ -35,14 +38,21 @@ class GraphOptimizationApplication(OptimizationApplication):
     An abstract class for graph optimization applications.
     """
 
-    def __init__(self, graph: Union[nx.Graph, np.ndarray, List]) -> None:
+    def __init__(self, graph: Union[rx.PyGraph, nx.Graph, np.ndarray, List]) -> None:
         """
         Args:
             graph: A graph representing a problem. It can be specified directly as a
             NetworkX Graph, or as an array or list if format suitable to build out a NetworkX graph.
         """
         # The view of the graph is stored which means the graph can not be changed.
-        self._graph = nx.Graph(graph).copy(as_view=True)
+        if isinstance(graph, rx.PyGraph):
+            self._graph = graph.copy()
+        elif isinstance(graph, nx.Graph):
+            self._graph = rx.networkx_converter(graph)
+        elif isinstance(graph, (np.ndarray, List)):
+            self._graph = rx.PyGraph.from_adjacency_matrix(graph)
```

Comment:
The input needs to be a numpy array here (mostly because the rust code in retworkx is working directly with the C array numpy points to), so if a list is allowed you'll need something like:

```suggestion
            self._graph = rx.PyGraph.from_adjacency_matrix(np.asarray(graph))
```

Reply:
Thanks!